### PR TITLE
New QAS Updates

### PR DESCRIPTION
**Pull request by edwardstamps**

### CODE REVIEW

File: mobile/src/home-view.js [hunk 1]
```diff
@@ -244,25 +287,33 @@ class HomeView extends Component {
...
+
     closeConfirm = () => {
-      this.setState({questionAsk: false})
+      this.setState({questionAsk: false, approveVar: false})
     }
 
     originalOrder = (questions) => {
       if (this.state.showRecent === false) {
-        questions.sort(function (a,b){
+        this.dateSort(questions)
```
Comment (AdamLiechty) [inline mobile/src/home-view.js]: This looks like it is sorting by date, then re-sorting by score immediately afterward.

File: mobile/src/home-view.js [hunk 2]
```diff
@@ -279,16 +330,22 @@ class HomeView extends Component {
     if (questionName.length === 0) {
       this.setState({showError: "red"})
     }
+    var approveVar = true
```
Comment (AdamLiechty) [inline mobile/src/home-view.js]: prefer `let` (or `const` when the value will not change) to `var`.

File: mobile/src/modal.js [hunk 3]
```diff
@@ -43,14 +42,14 @@ export class CustomModal extends Component {
...
-            marginTop: this.state.marginTop
+            height: this.state.height,
+            marginTop: 20,
+            paddingTop: 0,
         }
         
-        const androidStyle = {
-          paddingTop: 2,
-          paddingLeft: 0
+        const androidStyle1 = {
```
Comment (AdamLiechty) [inline mobile/src/modal.js]: is this name change a typo?

File: web/admin/src/App.css [hunk 4]
```diff
@@ -362,6 +363,16 @@
   border-radius: 50%;
 }
 
+.button3{
```
Comment (AdamLiechty) [inline web/admin/src/App.css]: Let's use a more meaningful class name.

File: web/admin/src/editCell.js [hunk 5]
```diff
@@ -84,25 +94,25 @@ export class CellEdit extends Component {
         var status = true
         if (named) {
             for (var item of this.props.sessions){
-                if (item.sessionName === named){
-                  status = false
-                  if (item.sessionName === this.props.task.sessionName){
-                      status = true
-                  }
+              if (item.sessionName.toUpperCase() === named.toUpperCase()){
+                status = false
```
Comment (AdamLiechty) [inline web/admin/src/editCell.js]: This is another place where we can replace these 4 lines with a single line: `status = (item.sessionName === this.props.task.sessionName)`

File: mobile/src/home-view.js [hunk 6]
```diff
@@ -38,7 +38,8 @@ class HomeView extends Component {
       title: "Q & A",
       questionAsk: false,
       questionError: "Ask Question",
-      topBorder: "#EFEFEF"
+      topBorder: "#EFEFEF",
+      approveVar: false
```
Comment (AdamLiechty) [inline mobile/src/home-view.js]: The `Var` is superfluous

File: mobile/src/home-view.js [hunk 7]
```diff
@@ -56,6 +57,25 @@ class HomeView extends Component {
...
+      sessRef.on('child_changed', data => {
+        var sessions = this.state.sessions
+        for (var i in sessions) {
+          if (sessions[i].key === data.key) {
+            sessions[i].sessionName = data.val().sessionName
+            if (this.state.session.key === data.key) {
+              var newSession = this.state.session
+              newSession.sessionName = data.val().sessionName
+              this.setState({sessions, session: newSession })
+              break;
```
Comment (AdamLiechty) [inline mobile/src/home-view.js]: In both branches of this inner `if`, you `break;`, so just put the break after this inner `if`

File: mobile/src/home-view.js (hunk 1 above)
Comment (edwardstamps) [reply mobile/src/home-view.js]: Exactly it should first order by date then score. It should act as a tiebreaker for questions with the same votes?

File: mobile/src/modal.js (hunk 3 above)
Comment (edwardstamps) [reply mobile/src/modal.js]: Was testing 2 different style constants. Will update.

File: mobile/src/home-view.js (hunk 6 above)
Comment (edwardstamps) [reply mobile/src/home-view.js]: Changed for future update

File: mobile/src/home-view.js (hunk 7 above)
Comment (edwardstamps) [reply mobile/src/home-view.js]: updated

File: mobile/src/home-view.js (hunk 2 above)
Comment (edwardstamps) [reply mobile/src/home-view.js]: updated

File: web/admin/src/App.css (hunk 4 above)
Comment (edwardstamps) [reply web/admin/src/App.css]: Agreed as it's is an image within a button

File: web/admin/src/editCell.js (hunk 5 above)
Comment (edwardstamps) [reply web/admin/src/editCell.js]: Updated

**Pull request merged**